server/portal/libs/agave/models/systems/base.py

```diff
@@ -354,28 +354,31 @@ def update_role(self, username, role):
         )
 
     def test(self):
-        """Test system
+        """Test storage system system
 
-        .. todo::
-            As of 05/2018 this only tests storage systems
-            by doing a `files-listing` on it.
-            What is a good way to test exec systems?
+        Test is a file listing of the system. If a system is missing, TAPIS *should* return
+        a 404.  Note though, that TAPIS sometimes returns an error that isn't a 404.
+
+        Execution systems are not supported.
         """
-        result = 'FAIL'
         success = True
-        # if self.type == self.TYPES.STORAGE:
+        result = 'SUCCESS'
         try:
             self._ac.files.list(
                 systemId=self.id,
                 filePath=''
             )
-            result = 'SUCCESS'
         except HTTPError as err:
+            logger.info("Test of system '{}' failed. Listing of system returned: {}".format(self.id, str(err)))
             success = False
-            result = err.response.json()
-        # elif self.type == self.TYPES.EXECUTION:
-        #     result = 'SUCCESS'
-
+            try:
+                result = err.response.json()
+            except JSONDecodeError:
+                result = 'FAIL'
+        except RequestException as err:
+            logger.error("Test of system '{}' failed unexpectedly! Listing of system returned: {}".format(self.id, str(err)))
+            success = False
+            result = "FAIL"
```

Comment:
Alternatively, we could possibly raise the exception. 

I'm starting to think that might be the best thing as we really aren't expecting this situation and if it occurs, it doesn't really imply the the storage system "failed" but that something bigger failed.

Reply:
I think that's a good idea, let's raise this exception